[Hazmat Shipping] Introduce Hazmat section with Feature Flag

ThomazFB opened the pull request:
Closes #10551 and partially fix #10583

Why
==========
Some companies like USPS are starting to require that merchants declare when a hazardous material is inside shipped packages. As the mobile app contains the shipping label creation feature, we need to comply with this requirement in order to avoid issues between our merchants and package shipping companies. 

This PR introduces the first step of the UI section responsible to allow the merchant to declare hazmat shipping.

How
==========
Introduce a UI section that's not connected to no models or network operation yet, just the UI section itself with the visibility bound by a respective feature flag. The instructions area of the section doesn't contain the expected clickable links yet, and the `Select a category` button is still pending. All this will become available in the following PRs for better breakdown of reviews.

Screen Capture
==========
https://github.com/woocommerce/woocommerce-ios/assets/5920403/75bc7928-922a-4d44-9b67-66b70822c08f

How to Test
==========
1. Open the order details of an order eligible for Shipping label creation
2. Start the shipping label creation flow until you reach the Package details section
3. Inside the package details view, make sure the Contains hazardous materials toggle is visible and interactable
4. Make sure the toggle expands and collapses a basic view that will serve as the main control options for the HAZMAT declaration

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Add initial hazmat section UI function to ShippingLabelSinglePackage

Commit:
Adjust string localizations inside ShippingLabelSinglePackage

Commit:
Introduce hazmat feature flag

ThomazFB committed:
Update Single Package View Model to contain published value for hazmat toggle

ThomazFB committed:
Introduce hazmat section basic layout

ThomazFB committed:
Introduce hazmat feature flag handling inside Single Package ViewModel

ThomazFB committed:
Implement visibility controls inside hazmat section

ThomazFB committed:
Add all hazmat instructions string resources

ThomazFB committed:
Reorganize SinglePackage UI elements in root

ThomazFB committed:
Introduce hazmat selection state handling

ThomazFB committed:
Refactor Hazmat section UI

ThomazFB committed:
FInish UI improvements over Hazmat section

ThomazFB committed:
Fix lint issues

ThomazFB committed:
Rename SwiftUI Hazmat functions

ThomazFB committed:
Fix more lint issues

ThomazFB committed:
Add missing text alignments inside Hazmat UI

ThomazFB committed:
Final adjustments on Hazmat UI strings

ThomazFB committed:
Add hazmat toggle tooltip conditional rendering

ThomazFB committed:
Merge branch 'trunk' into issue/10551-create-hazmat-checkbox

jaclync commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelSinglePackage.swift:
```diff
@@ -90,29 +91,95 @@ struct ShippingLabelSinglePackage: View {
+        }
+    }
+
+    func createHazmatSection() -> some View {
```
nit: It might be nice moving some of the Hazmat view code to a separate SwiftUI view, so that the details of the UI (like the long text for instructions) don't have to be in this main view.

jaclync reviewed (approved):
LGTM! :shipit: Just a non-blocking suggestion to move the Hazmat UI code to a separate view.

Non-blocking UI observation: the footer of the package weight section now looks like the header of the Hazmat section (not sure if Android has this footer as well):

<img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/f880a3b0-3972-46ef-86da-b76c752bb07a" width="300" />

Maybe the insets can be updated so that the footer text can be closer to the section above?

ThomazFB committed:
Define HazmatDeclaration file

ThomazFB committed:
Add HazmatDeclaration content

ThomazFB committed:
Remove Hazmat related UI from SinglePackage UI file

ThomazFB committed:
Rename HazmatDeclaration file to SinglePackageHazmatDeclaration

ThomazFB committed:
Merge branch 'trunk' into issue/10551-create-hazmat-checkbox

ThomazFB committed:
Improve package weight footnote spacing

ThomazFB commented:
Thank you @jaclync for your review! I've addressed all your comments and the PR is ready for a second round!

>It might be nice moving some of the Hazmat view code to a separate SwiftUI view, so that the details of the UI (like the long text for instructions) don't have to be in this main view.

True! It's fixed now, I defined a `SinglePackageHazmatDeclaration` SwiftUI type that now holds everything related to the hazmat operations and receives the same ViewModel through the `init` call.

>the footer of the package weight section now looks like the header of the Hazmat section (not sure if Android has this footer as well)

Good catch! I've noticed that this footer was using the `ListHeaderView`, which seems a bit odd because that component is enforcing a big top padding meant for the header of a list instead of a footer. I replaced it with a common `Text` using the `footnoteStyle` and proper paddings for this footer scenario. The change is introduced in https://github.com/woocommerce/woocommerce-ios/pull/10552/commits/806254d4b9ca716f9206f2cfc05e7b2d4d773309. Let me know what you think!

ThomazFB committed:
Add if debug constraint to Hazmat section preview

jaclync commented on WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/SinglePackageHazmatDeclaration.swift:
```diff
@@ -0,0 +1,127 @@
+        .renderedIf(viewModel.isHazmatShippingEnabled)
+    }
+
+    func createHazmatInstructionsView() -> some View {
```
super nit: this function can be private like in a private extension

ThomazFB replied:
Oh, thank you for the suggestion! I thought that View returning functions didn't required visibility changes, I'll make sure to fix that for the other SwiftUI functions I declared. Thanks! Fixed in https://github.com/woocommerce/woocommerce-ios/pull/10552/commits/940b3c8d7c89e8e488a309c679a9bd0d92e75f46

jaclync reviewed (approved):
Thanks for the updates, LGTM! 👍 

Can leave it for a future improvement on the landscape mode (probably updating the padding for the views to account for safe area insets):

<img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/0e7a633a-7d33-4a8c-9169-62eb87f9d9b6" width="300" />

ThomazFB committed:
Adjust use of safeAreaInsets inside Hazmat section

ThomazFB committed:
Adjust visibility modifiers of HazmatDeclaration section

ThomazFB commented:
Thank you @jaclync for this careful UI check 🙇, I need to improve my eye for UI testing when coding in iOS environment 😬. I  adjusted the safe insets usage and now the section is following the same UI approach of the view when in landscape. https://github.com/woocommerce/woocommerce-ios/pull/10552/commits/58395fde1868d640199921d318ce5fe700946e0d
